interfaces/repo_test.go

```diff
@@ -1239,3 +1239,85 @@ func (s *RepositorySuite) TestAutoConnectContentInterfaceNoMatchingDeveloper(c *
 	candidateSlots := repo.AutoConnectCandidates("content-plug-snap", "import-content")
 	c.Check(candidateSlots, HasLen, 0)
 }
+
+func makeLivepatchConnectionTestSnaps(c *C, name, developer string) (*Repository, *snap.Info, *snap.Info) {
+	repo := NewRepository()
+	err := repo.AddInterface(&TestInterface{InterfaceName: "restricted", AutoConnectFlag: false})
+	c.Assert(err, IsNil)
+
+	err = repo.AddInterface(&TestInterface{InterfaceName: "non-restricted", AutoConnectFlag: true})
+	c.Assert(err, IsNil)
+
+	plugSnap, err := snap.InfoFromSnapYaml([]byte(fmt.Sprintf(`
+name: %s
+plugs:
+  canonical-livepatch:
+    interface: restricted
+  non-restricted:
+    interface: non-restricted
+`, name)))
+	c.Assert(err, IsNil)
+	slotSnap, err := snap.InfoFromSnapYaml([]byte(`
+name: ubuntu-core
+type: os
+slots:
+  restricted:
+    interface: restricted
+  non-restricted:
+    interface: non-restricted
+`))
+	c.Assert(err, IsNil)
+
+	plugSnap.DeveloperID = developer
+	slotSnap.DeveloperID = "canonical"
+	slotSnap.Type = snap.TypeOS
+
+	err = repo.AddSnap(plugSnap)
+	c.Assert(err, IsNil)
+	err = repo.AddSnap(slotSnap)
+	c.Assert(err, IsNil)
+
+	return repo, plugSnap, slotSnap
+}
+
+// test auto-connecting livepatch interfaces for special snaps
+func (s *RepositorySuite) TestAutoConnectLivepatchInterfaces(c *C) {
+	repo, _, _ := makeLivepatchConnectionTestSnaps(c, "canonical-livepatch", "canonical")
+	candidateSlots := repo.AutoConnectCandidates("canonical-livepatch", "canonical-livepatch")
```

Comment:
I didn't notice this before, but naming the test plug interface 'canonical-livepatch' and having the plugging snap's name 'canonical-livepatch' is confusing when verifying the correctness of these tests. Perhaps rename the plugged interface to 'restricted-interface' (keeping 'type: restricted' since you seem to want to test with the reference and type as different).